packages/web/src/hooks/useAccountTransition.ts

```diff
@@ -0,0 +1,121 @@
+import { useEffect, useRef, useState, useCallback, useMemo } from 'react'
+
+import { accountSelectors } from '@audius/common/store'
+
+import { useSelector } from 'utils/reducer'
+
+const { getUserHandle, getUserId, getIsAccountComplete } = accountSelectors
+
+const PREVIOUS_ACCOUNT_STATE_KEY = 'previousAccountState'
+
+type AccountTransitionState = {
+  userId: number | null
+  handle: string | null
+  isComplete: boolean
+}
+
+/**
+ * Reads the previous account state from localStorage
+ */
+const previousAccountState = (): AccountTransitionState | null => {
+  try {
+    const savedState = window.localStorage.getItem(PREVIOUS_ACCOUNT_STATE_KEY)
```

Comment:
We have a localStorage adapter for this stuff, prefer using that if possible.
There's also a constants file that has the localStorage keys in it.